pkg/chunk/cache/memcached_client.go

```diff
@@ -98,8 +98,10 @@ func NewMemcachedClient(cfg MemcachedClientConfig, name string) MemcachedClient
 		hostname:   cfg.Host,
 		service:    cfg.Service,
 		addresses:  strings.Split(cfg.Addresses, ","),
-		provider:   dns.NewProvider(util.Logger, prometheus.DefaultRegisterer, dns.GolangResolverType),
-		quit:       make(chan struct{}),
+		provider: dns.NewProvider(util.Logger, prometheus.WrapRegistererWith(prometheus.Labels{
+			"name": name,
```

Comment:
Is `name` guaranteed to be unique across the codebase? I've seen we're using `cfg.Prefix` as name (which looks another bad smell to me because it's not a name) so I would like to make sure uniqueness is guaranteed.

Reply:
Yes it should be, as multiple registrations of the prefix would be caught by flag registration.